app/controllers/application_controller.rb

```diff
@@ -30,17 +30,22 @@ def decorated_current_user
 
   #TODO: move this into a Warden after_authentication method
   # This seems like a stop-gap but works technically
-  def complete_two_factor_setup
-    whitelist = [
-      destroy_user_session_path
-    ]
-
-    if current_user && !current_user.account_verified && current_user.confirmed_at
-      # Users need to be able to verify their tfa code
-      unless request.path.start_with?(users_two_factor_setup_path) ||
-        whitelist.include?(request.path)
-
-        redirect_to new_users_two_factor_setup_path
+  if Rails.env.development?
```

Comment:
Should probably also check for `Rails.configuration.force_2fa.present?`. If it's present, you should do the 2fa even in dev.